### PR TITLE
Adding support for frontend HTTPS with pre shared certs

### PR DESCRIPTION
**Pull request by nikhiljindal**
Ref https://github.com/GoogleCloudPlatform/k8s-multicluster-ingress/issues/46

Adding support for frontend HTTPS  with pre shared certs

cc @G-Harmon @csbell @madhusudancs

### CODE REVIEW

**Comment by nikhiljindal** [comment]
friendly ping

File: app/kubemci/pkg/gcp/sslcert/sslcertsyncer.go [hunk 1]
```diff
@@ -114,6 +124,13 @@ func (s *SSLCertSyncer) updateSSLCert(desiredCert *compute.SslCertificate) (stri
 	name := desiredCert.Name
 	fmt.Println("Deleting existing ssl cert", name, "and recreating it to match the desired state.")
 	// SSL Cert does not support update. We need to delete and then create again.
+	// Note: This is different than what we do in ingress-gce.
+	// In ingress-gce, we first create a new cert with a different name,
+	// update the target proxy to point to the new cert and then delete
+	// the old cert.
+	// What we do here is simpler, but can lead to downtime in the brief period
+	// when we have deleted the old cert but havent created the new one.
+	// TODO(nikhiljindal): Converge this with ingress-gce by sharing the same code.
```
Comment (G-Harmon) [inline app/kubemci/pkg/gcp/sslcert/sslcertsyncer.go]: file a bug for this?

File: app/kubemci/pkg/gcp/sslcert/sslcertsyncer_test.go [hunk 2]
```diff
@@ -100,25 +105,70 @@ func TestEnsureSSLCert(t *testing.T) {
...
-				},
-			},
 			Backend: &v1beta1.IngressBackend{
 				ServiceName: svcName,
 				ServicePort: intstr.FromString(svcPort),
 			},
 		},
 	}
 	client := &fake.Clientset{}
+	if !withSecret {
```
Comment (G-Harmon) [inline app/kubemci/pkg/gcp/sslcert/sslcertsyncer_test.go]: nit: I usually like early returns, but this feels like it's setting us up for someone to forget that there was an early return, and add code at the end of the function that doesn't always run when it should. thoughts?

Comment (nikhiljindal) [reply app/kubemci/pkg/gcp/sslcert/sslcertsyncer_test.go]: right. Refactored the secret part into a separate method

File: app/kubemci/pkg/gcp/sslcert/sslcertsyncer.go (hunk 1 above)
Comment (nikhiljindal) [reply app/kubemci/pkg/gcp/sslcert/sslcertsyncer.go]: Done https://github.com/GoogleCloudPlatform/k8s-multicluster-ingress/issues/124

**Comment by nikhiljindal** [comment]
Thanks for the review @G-Harmon 

Regarding coverage: its 50% because ive tested 9 lines out of 18.
Out of those 9 lines, 7 are comments and 2 are package includes :)

#TIL deleting comments can increase code coverage :)

**Comment by G-Harmon** [comment]
LGTM.

> #TIL deleting comments can increase code coverage :)

heh, that's not a very smart coverage tool, then!
But don't you test EnsureSSLCert? Seems like that should give you a higher coverage result?

**Comment by nikhiljindal** [comment]
> But don't you test EnsureSSLCert? Seems like that should give you a higher coverage result?

That 50% is only for the lines that I am adding in this PR. I am testing 9 of the 18 lines that am adding in this PR.

**Pull request merged**